auto_submit/lib/service/validation_service.dart

```diff
@@ -146,20 +145,112 @@ class ValidationService {
       }
     }
     // If we got to this point it means we are ready to submit the PR.
-    bool processed = await processMerge(config, result, messagePullRequest);
-    if (processed) await pubsub.acknowledge('auto-submit-queue-sub', ackId);
-    log.info('Ack the processed message : $ackId.');
+    await processMergeSafely(
+        config, gitHubService, messagePullRequest, result, pubsub, ackId, slug, prNumber, Config.kAutosubmitLabel);
+  }
+
+  /// The logic for processing a revert request and opening the follow up
+  /// review issue in github.
+  Future<void> processRevertRequest(
+    Config config,
+    QueryResult result,
+    gh.PullRequest messagePullRequest,
+    String ackId,
+    PubSub pubsub,
+  ) async {
+    ValidationResult revertValidationResult = await revertValidation!.validate(result, messagePullRequest);
+
+    gh.RepositorySlug slug = messagePullRequest.base!.repo!.slug();
+    final int prNumber = messagePullRequest.number!;
+    final GithubService githubService = await config.createGithubService(slug);
+
+    if (revertValidationResult.result) {
+      try {
+        bool processed = await processMerge(config, result, messagePullRequest);
+        if (processed) {
+          gh.Issue issue = await githubService.createIssue(
+            slug,
+            'Follow up review for revert pull request $prNumber',
+            'Revert request by author ${result.repository!.pullRequest!.author}',
+          );
+          log.info('Issue #${issue.id} was created to track the review for $prNumber in ${slug.fullName}');
+        } else {
+          String message = 'Unable to merge pull request $prNumber.';
+          log.warning(message);
+          await removeLabelAndComment(githubService, slug, prNumber, Config.kRevertLabel, message);
+        }
+      } catch (exception) {
```

Comment:
Prefer specifying the type to catch, instead of a generic catch all.

Would this be a GitHubError exception?

Reply:
Yeah that is one of the cons of dart, the fact that you need to guess or go dig through code when a throws declaration would could be immensely helpful. /done ranting. I will look for the explicit exception.

Reply:
Added catch for explicit GithubError and Exception as they are both thrown.

Reply:
Everything thrown in Dart should be a subtype of Exception. Is there a specific exception you're trying to catch?

Reply:
Yes but Exception is thrown when we are calling createGitHubGraphQLClient(...) and if we do not have permissions our method in config.dart for _generateGithubToken will throw an exception.
 https://github.com/flutter/cocoon/blob/main/auto_submit/lib/service/config.dart#L128
But I see a way to rework this now from this conversation.